### PR TITLE
add verify domain call

### PR DESCRIPTION
**Pull request by mwotton**

### CODE REVIEW

File: src/Network/API/Mandrill/Senders.hs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+verifyDomain k email =
+  toMandrillResponse VerifyDomain
+  (VerifyDomainRq k (decodeUtf8 $ TEV.domainPart email) (decodeUtf8 $ TEV.localPart email))
+-- addDomain :: MandrillKey
+
+-- addDomain k dom = toMandrillResponse DomainsAdd (DomainAddRq k dom)
+
+
+
+-- addRoute :: MandrillKey
```
Comment (adinapoli) [inline src/Network/API/Mandrill/Senders.hs]: Shall we simply avoid committing commented code? 😀

**Comment by adinapoli** [comment]
Thanks @mwotton , looks great! (apart from a very minor remark I added).

Shall we be good open source citizens and also expand the `README` section of the supported api calls ? (the list at the bottom of the page).

Thanks!

A.

**Comment by mwotton** [comment]
Haha oops :) yeah, shall do.

**Comment by mwotton** [comment]
btw, you may be interested in checking out servant-client for your next project - it has the same sort of approach while removing some of the boilerplate.

**Comment by adinapoli** [comment]
> btw, you may be interested in checking out servant-client for your next project

Yes! `servant-client` and the whole `servant` family it's pretty sweet, although I'm waiting for it to become a bit more stable before adopting it. Do you have any experience using it? I was tempted using servant at work in the past at least twice, but was put off by the aforementioned; it's seems to still be very much in the flux!

**Comment by mwotton** [comment]
The big sticking point is auth for me at the moment. I've got something up at github.com/mwotton/dnsmadeeasy - it works, but you don't have a monad instance for running compound things easily at the moment if you also need auth. Very promising, though.

**Comment by adinapoli** [comment]
> The big sticking point is auth for me at the moment.

Yes, I heard that auth was still under active dev. Hopefully they will sort it out relatively quickly, as I can imagine it's quite important for RESTful services :D 

PRwis it looks great, will merge as soon as Travis will give us green light.

Thank you!

**Pull request merged**